plugin/plugin.go

```diff
@@ -0,0 +1,209 @@
+package plugin
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+
+	"github.com/PagerDuty/go-pagerduty"
+	"github.com/sirupsen/logrus"
+)
+
+// Severity levels as constants for reuse and readability.
+const (
+	SeverityCritical = "critical"
+	SeverityError    = "error"
+	SeverityWarning  = "warning"
+	SeverityInfo     = "info"
+	SeverityUnknown  = "unknown"
+)
+
+// Args provides plugin execution arguments.
+type Args struct {
+	Level             string `envconfig:"PLUGIN_LOG_LEVEL"`
+	RoutingKey        string `envconfig:"PLUGIN_ROUTING_KEY"`
+	IncidentSummary   string `envconfig:"PLUGIN_INCIDENT_SUMMARY"`
+	IncidentSource    string `envconfig:"PLUGIN_INCIDENT_SOURCE"`
+	IncidentSeverity  string `envconfig:"PLUGIN_INCIDENT_SEVERITY"`
+	DedupKey          string `envconfig:"PLUGIN_DEDUP_KEY"`
+	CreateChangeEvent bool   `envconfig:"PLUGIN_CREATE_CHANGE_EVENT"`
+	Resolve           bool   `envconfig:"PLUGIN_RESOLVE"`
+	JobStatus         string `envconfig:"PLUGIN_JOB_STATUS"`
+	CustomDetailsStr  string `envconfig:"PLUGIN_CUSTOM_DETAILS"` // Intermediate string to receive JSON
+	CustomDetails     map[string]interface{}
+}
+
+// PagerDutyClient defines the methods used from the PagerDuty API.
+type PagerDutyClient interface {
+	ManageEventWithContext(ctx context.Context, event *pagerduty.V2Event) (*pagerduty.V2EventResponse, error)
+	CreateChangeEventWithContext(ctx context.Context, event pagerduty.ChangeEvent) (*pagerduty.ChangeEventResponse, error)
+}
+
+// validateSeverity validates PagerDuty's allowed severity values.
+func validateSeverity(severity string) error {
+	switch severity {
+	case SeverityCritical, SeverityError, SeverityWarning, SeverityInfo, SeverityUnknown:
```

Comment:
`SeverityUnknown` isn’t used elsewhere in the code. remove it to avoid confusion